### ISSUE TITLE
[Sprint] Fix Keras Shuffling Leakage and Upgrade Autoencoder Regression Head

### ISSUE THREAD

**Issue by ankit02327** [ankit02327/stock-price#115]
As part of the Algorithmic & ML Sprint, we need to address time-series data leakage within the Autoencoder model (`backend/algorithms/optimised/autoencoders/autoencoder.py`) and replace its weak prediction head.

Currently, the Keras `fit()` method is called without specifying the `shuffle` parameter, meaning it defaults to `shuffle=True`. While the `validation_split` correctly isolates the last 20% of data chronologically, shuffling the remaining training data breaks the time-series sequence and leaks temporal patterns across batches. Additionally, the model utilizes a deep neural network to extract complex, non-linear encoded features, but then uses a simple `LinearRegression` model to make the final prediction. This creates a severe bottleneck that fails to capture the non-linear relationships discovered by the encoder.

**Tasks:**
- [ ] Explicitly set `shuffle=False` in the `self.autoencoder.fit()` call to maintain strict chronological order during time-series training.
- [ ] Replace the `LinearRegression` prediction head with a non-linear regressor (e.g., `MLPRegressor`, `SVR`, or `RandomForestRegressor`) to fully utilize the encoded features.
- [ ] Update the Keras model saving extensions from the legacy `.h5` format to the modern `.keras` format to prevent future deprecation crashes.

**Comment by ankit02327** [comment]
1. **Time-Series Integrity:** Add `shuffle=False` to the `self.autoencoder.fit()` arguments.
2. **Upgrade Regression Head:** Import `MLPRegressor` from `sklearn.neural_network` (or another non-linear model). In `fit()`, assign `self.regressor = MLPRegressor(hidden_layer_sizes=(32, 16), max_iter=500)` instead of `LinearRegression()`.
3. **Modernize I/O:** In the `save()` and `load()` methods, change the file extensions from `_autoencoder.h5` and `_encoder.h5` to `_autoencoder.keras` and `_encoder.keras`.

**Comment by ankit02327** [comment]
```py
# 1. Maintain chronological order during batch training
history = self.autoencoder.fit(
    X_scaled, X_scaled,
    epochs=100,
    batch_size=32,
    validation_split=0.2,
    callbacks=callbacks,
    verbose=1,
    shuffle=False  # CRITICAL: Prevent time-series data leakage
)

# 2. Upgrade the regression head to capture non-linear relationships
from sklearn.neural_network import MLPRegressor

logger.info("Training non-linear regression model on encoded features...")
self.regressor = MLPRegressor(
    hidden_layer_sizes=(32, 16), 
    activation='relu', 
    max_iter=1000, 
    early_stopping=True,
    random_state=42
)
self.regressor.fit(encoded_features, y)

# 3. Modernize Keras saves
self.autoencoder.save(f"{path}_autoencoder.keras")
self.encoder.save(f"{path}_encoder.keras")
```

**Comment by AKASHPANDEY4642** [comment]
I got an idea, I'll submit a pr soon

**Comment by ankit02327** [comment]
Would love to look at your approach. I am assigning this issue to your for now. @AKASHPANDEY4642
